### PR TITLE
test: add tests to check favourites endpoints

### PR DESCRIPTION
**Pull request by rubvalave**
-Adds missing tests and helpers for favourites endpoints tests.

### CODE REVIEW

File: app/test/e2e/favourite/favourite-delete.spec.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        await Favourite.deleteMany({}).exec();
+    });
+
+    it('Deleting a favourite while not being authenticated should return a 401', async () => {
+        const favourite = await (new Favourite(createFavourite({ userId: USERS.USER.id }))).save();
+
+        const response = await requester
+            .delete(`/api/v1/favourite/${favourite._id}`);
+
+        ensureCorrectError(response.body, 'Unauthorized');
```
Comment (tiagojsag) [inline app/test/e2e/favourite/favourite-delete.spec.js:40]: are these validating the http response code? I think you have a few of those missing

File: app/test/e2e/utils/helpers.js [hunk 2]
```diff
@@ -49,6 +49,17 @@ const createCollection = (additionalData = {}) => {
     };
 };
 
+const createFavourite = (additionalData = {}) => {
+    const fakeResourceId = getUUID();
+    return {
+        resourceId: fakeResourceId,
+        resourceType: additionalData.resourceType || 'dataset',
+        application: 'rw',
+        createdAt: '2017-12-27T16:09:46.855Z',
```
Comment (tiagojsag) [inline app/test/e2e/utils/helpers.js]: any chance this can be removed? it should be autofilled

File: app/test/e2e/utils/helpers.js [hunk 3]
```diff
@@ -329,6 +340,18 @@ const mockDeleteGraphAssociation = (datasetId, application = 'rw', mockSuccess =
         .reply(mockSuccess ? 200 : 404, { data: mockSuccess ? {} : { message: 'Resource XXXX not found.' } });
 };
 
+const mockAddFavouriteResourceToGraph = (resourceType, resourceId, loggedUser) => {
+    nock(process.env.GATEWAY_URL)
+        .post(`/v1/graph/favourite/${resourceType}/${resourceId}/${loggedUser.id}`,)
```
Comment (tiagojsag) [inline app/test/e2e/utils/helpers.js:343]: no body to validate?

**Review by tiagojsag — CHANGES_REQUESTED** [review]
a few minor things can be improved, otherwise LGTM

File: app/test/e2e/utils/helpers.js (hunk 3 above)
Comment (rubvalave) [reply app/test/e2e/utils/helpers.js:343]: I'm not sure about what the body should have. Nor the favourites or graph documentation in doc-api, nor the request for graph api inside favourite.router shows anything about what this body is supposed to return. The request is done but without expecting nothing in return or use the body for anything inside the code.

Is this how it is supposed to work, though?

**Pull request merged**